### PR TITLE
[2.0] Backport of #4700

### PR DESCRIPTION
**Pull request by dstansby**

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -1092,26 +1099,21 @@ def world_to_pixel(self, coordinate, origin=0):
...
+        x, y = self.wcs.world_to_pixel(coordinate)
+        if origin == 1:
+            x += 1
+            y += 1
 
         return PixelPair(x * u.pixel, y * u.pixel)
 
     @u.quantity_input
-    def pixel_to_world(self, x: u.pixel, y: u.pixel, origin=0):
+    def pixel_to_world(self, x: u.pixel, y: u.pixel, origin=None):
```
Comment (Cadair) [inline sunpy/map/mapbase.py:1111]:
```suggestion
    def pixel_to_world(self, x: u.pixel, y: u.pixel, origin=0):
```

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -1068,10 +1068,18 @@ def _validate_meta(self):
...
+
+    @staticmethod
+    def _check_origin(origin):
+        """
+        Check origin is valid.
+        """
+        if origin not in [None, 0, 1]:
+            raise ValueError('origin must be 0 or 1.')
+
+    def world_to_pixel(self, coordinate, origin=None):
```
Comment (Cadair) [inline sunpy/map/mapbase.py:1080]:
```suggestion
    def world_to_pixel(self, coordinate, origin=0):
```

**Review by Cadair — APPROVED** [review]
There is no reason to change the defaults right?

File: sunpy/map/mapbase.py (hunk 1 above)
Comment (dstansby) [reply sunpy/map/mapbase.py:1111]: Woops, I forgot to leave that out, thanks for catching.

**Comment by dstansby** [comment]
I fixed the comments, and also slimmed this down a little bit more by leaving the error message/check the same to reduce the changes.

**Review by Cadair — APPROVED** [review]

**Comment by Cadair** [comment]
Can you fix whatever codestyle is complaining about?

**Pull request merged**